packages/block-library/src/post-author/index.php

```diff
@@ -0,0 +1,33 @@
+<?php
+/**
+ * Server-side rendering of the `core/post-author` block.
+ *
+ * @package WordPress
+ */
+
+/**
+ * Renders the `core/post-author` block on the server.
+ *
+ * @return string Returns the filtered post author for the current post wrapped inside "h6" tags.
+ */
+function render_block_core_post_author() {
+	$post = gutenberg_get_post_from_context();
+	if ( ! $post ) {
+		return '';
+	}
+	// translators: %s: The author.
+	return sprintf( __( '<address>By %s</address>' ), get_the_author() );
```

Comment:
This is causing a lint error on WP.com when attempting to update the plugin there:

```
!! Please put the opening and closing HTML tags outside of the translatable string:
wp-content/plugins/gutenberg-core/7.3.0-alpha/build/block-library/blocks/post-author.php:19 __( "<address>By %s</address>" )
```

which seems fair. @chriskmnds is working on a fix.

If core contributors feel that a lint check this makes sense, is there any chance we can add one to CI?

/cc @akirk

Reply:
Why is this an issue? It would just make the code harder to read with the concatenations to what end?

cc @aduth @youknowriad

Reply:
The suggestion is to have translators only translate `By %s`. For one, it lets the translator focus on the text. Secondly, there is possible translation reuse because `By %s` has already been transalted. Finally, if the HTML tags are added through code, it means you can change them later to add CSS classes, ids, whatnot, without having to change the translations.

Reply:
It makes easier for translators because the html tag in that case will just be repeated. It doesn't make sense to allow translators to change the HTML tag.

Reply:
That makes sense.

Should I open a PR or are you already on it?

Reply:
I have a fix, but putting together the PR properly, testing, etc. will take a while (even for something simple as this). :D

Reply:
https://github.com/WordPress/gutenberg/pull/19675

will test tomorrow, but feel free to review pls :-)

Reply:
For posterity's sake, there is a follow-up task related to this conversation at: #19911